lib/ansible/parsing/dataloader.py

```diff
@@ -233,35 +236,33 @@ def path_dwim(self, given):
         given = unquote(given)
         given = to_text(given, errors='surrogate_or_strict')
 
-        if given.startswith(u"/"):
-            return os.path.abspath(given)
-        elif given.startswith(u"~"):
-            return os.path.abspath(os.path.expanduser(given))
+        if RE_ABS_PATH.search(given):
+            path = unfrackpath(given)
         else:
             basedir = to_text(self._basedir, errors='surrogate_or_strict')
-            return os.path.abspath(os.path.join(basedir, given))
+            path = unfrackpath(os.path.join(basedir, given))
+
+        return path
```

Comment:
Are we sure we want path_dwim to call unfrackpath?  That changes the semantics of path_dwim and could break code if it relied on symlinks not being derefernced (for instance).  Perhaps path_dwim() should call unfrackpath with follow=False?